### PR TITLE
[stat var hierarchy] group search results

### PR DESCRIPTION
**Pull request by chejennifer**
- group stat var search results by stat var group: If a stat var matches the tokens and has a parent stat var group that also matches the tokens, nest the stat var result under the stat var group result.

### CODE REVIEW

File: proto/stat_var.proto [hunk 1]
```diff
@@ -181,10 +181,19 @@ message SearchStatVarRequest {
   bool enable_blocklist = 3;
 }
 message SearchStatVarResponse {
```
Comment (shifucun) [inline proto/stat_var.proto:183]: Nit: can move SearchResultSVG one level up, so it's easier to reference in the code

File: internal/server/statvar/statvar_search.go [hunk 2]
```diff
@@ -105,6 +124,43 @@ func filter(
...
+		}
+	}
+	resultSv := []*pb.EntityInfo{}
+	// Iterate through the list of stat vars. If a stat var has a parent svg that
+	// is a result, add the stat var to that svg result. If not, add the stat var
+	// to the result list of stat vars
+	for _, sv := range svList {
+		isGrouped := false
+		if parents, ok := parentMap[sv.Dcid]; ok {
+			for _, parent := range parents {
```
Comment (shifucun) [inline internal/server/statvar/statvar_search.go:144]: I wonder should sort the parents here so stat var is always added to the top parent.

**Review by shifucun — APPROVED** [review]
Very nice change! The response is much better read now.

File: proto/stat_var.proto (hunk 1 above)
Comment (chejennifer) [reply proto/stat_var.proto:192]: done

File: internal/server/statvar/statvar_search.go (hunk 2 above)
Comment (chejennifer) [reply internal/server/statvar/statvar_search.go:151]: I think that's a good idea, done

**Review by beets — APPROVED** [review]
this is nice!

**Comment by chejennifer** [comment]
thanks!

**Pull request merged**